### PR TITLE
core: support user manager with Condition[Memory/CPU/IO]Pressure

### PR DESCRIPTION
**Pull request by bluca**
Get the cgroup root path from the current PID, so that when
ran by the user manager we can get to the right path.
Eg: foo.slice:10% will check under:

/sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/foo.slice/cpu.pressure

Follow-up for 81513b382b24a7f3602987f71042d075ca27d1a5

### CODE REVIEW

File: src/shared/condition.c [hunk 1]
```diff
@@ -992,12 +992,12 @@ static int condition_test_psi(Condition *c, char **env) {
         if (r == 1) {
                 pressure_path = path_join("/proc/pressure", pressure_type);
                 if (!pressure_path)
-                        return log_oom();
+                        return -ENOMEM;
```
Comment (bluca) [inline src/shared/condition.c]: we log at debug level here, so log_oom was wrong too

Comment (yuwata) [reply src/shared/condition.c]: Then, `log_oom_debug()`?

Comment (bluca) [reply src/shared/condition.c]: sure, done

**Comment by yuwata** [comment]
About the commit message. #21591 -> #21437

**Comment by bluca** [comment]
> About the commit message. #21591 -> #21437 Also, #21437 contains only one commit, so it is better to mention the commit hash, instead of the PR number.

oops had the wrong tab open - fixed

**Comment by bluca** [comment]
looks like in some cases (test running in packit) we get empty or `/` from `cg_pid_get_path` - small adjustment to deal with that

File: src/shared/condition.c [hunk 2]
```diff
@@ -1030,7 +1030,22 @@ static int condition_test_psi(Condition *c, char **env) {
...
+
+                /* Drop init.scope, we want the parent. We could get an empty or / path, but that's fine,
+                 * avoid failing and simply path_join with an empty string, which will do the right thing. */
+                r = path_extract_directory(root_scope, &root_path);
+                if (r < 0 && r != -EADDRNOTAVAIL)
+                        return log_debug_errno(r, "Failed to extract parent directory of '%s': %m", root_scope);
+
+                slice_joined = path_join(strempty(root_path), slice_path);
+                if (!slice_joined)
+                        return log_oom_debug();
```
Comment (yuwata) [inline src/shared/condition.c]: How about the following?
```
if (!empty_or_root(root_scope)) {
        _cleanup_free_ char *root_path = NULL, *slice_joined = NULL;

        r = path_extract_directory(root_scope, &root_path);
        if (r < 0)
                return log_debug_errno(r, ...);

        slice_joined = path_join(root_path, slice_path);
        if (!slice_joined)
                return log_oom_debug();

        free_and_replace(slice_path, slice_joined);
}

r = cg_get_path(..., slice_path, ...);
```

Comment (bluca) [reply src/shared/condition.c]: sure, applied

**Comment by Werkov** [comment]
Ad tests -- Why is not the `manager_setup_cgroup()` affected the same way? Is it because of `path_extract_directory()` failing on those inputs? It'd be more consistent to have same approach (`endswith(m->cgroup_root, "/" SPECIAL_INIT_SCOPE);`) in all places.

**Comment by bluca** [comment]
> Ad tests -- Why is not the `manager_setup_cgroup()` affected the same way? Is it because of `path_extract_directory()` failing on those inputs? It'd be more consistent to have same approach (`endswith(m->cgroup_root, "/" SPECIAL_INIT_SCOPE);`) in all places.

ok, updated

**Pull request merged**